Core/include/Acts/Seeding/GNN_DataStorage.hpp

```diff
@@ -0,0 +1,151 @@
+#include<vector>
+#include<map>
+#include<algorithm>
+
+#define MAX_SEG_PER_NODE 1000 //was 30
+#define N_SEG_CONNS  6 //was 6
+
+#include "Acts/Seeding/GNN_Geometry.h"
+
+namespace Acts {
+template <typename space_point_t>  
+class TrigFTF_GNN_Node {
+ public:
+
+  struct CompareByPhi {
+
+    bool operator()(const TrigFTF_GNN_Node<space_point_t>*  n1, const TrigFTF_GNN_Node<space_point_t>*  n2) {
+      return n1->m_sp.phi() < n2->m_sp.phi();
+    }
+
+  };
+  //want constructor to take simspace point 
+  TrigFTF_GNN_Node(const space_point_t&, float, float);  
+  ~TrigFTF_GNN_Node();
+  
+
+  inline void addIn(int i) {
+      if(m_in.size()<MAX_SEG_PER_NODE) {
+       m_in.push_back(i);
+     }
+  }
+
+  inline void addOut(int i) {
+      if(m_out.size()<MAX_SEG_PER_NODE) {
+        m_out.push_back(i);
+      }
+  }
+  
+  inline bool isConnector() const {
+      if(m_in.empty() || m_out.empty()) return false;
+      return true;
+  }
+
+  inline bool isFull() const {
+      if(m_in.size()==MAX_SEG_PER_NODE && m_out.size()==MAX_SEG_PER_NODE) return true;
+      else return false;
+  }
+
+  const space_point_t& m_sp;
+  
+  std::vector<unsigned int> m_in;//indices of the edges in the edge storage
+  std::vector<unsigned int> m_out;
+  float m_minCutOnTau, m_maxCutOnTau;
+
+};
+
+template <typename space_point_t>  
+class TrigFTF_GNN_EtaBin {
+ public:
+  TrigFTF_GNN_EtaBin();
+  ~TrigFTF_GNN_EtaBin();
+
+  void 
+  hi();
+
+  bool empty() const {
+    return m_vn.empty();
+  }
+  
+  void generatePhiIndexing(float);
+  
+  std::vector<TrigFTF_GNN_Node<space_point_t>*> m_vn;
```

Comment:
```suggestion
  std::vector<std::unique_ptr<TrigFTF_GNN_Node<space_point_t>>> m_vn;
```
and then remove code for `delete`ing all the nodes.